File: 240517/방향에 맞춰 이동/move-in-direction.java
import java.io.*;
import java.util.*;

public class Main  {
    public static void main(String[] args)  {
        Scanner sc = new Scanner(System.in);
        int input = sc.nextInt();
        int x = 0;
        int y = 0;
        for (int i = 0; i <= input; i++) {
            String[] a = sc.nextLine().split(" ");
            if (a[0].equals("N")) {
                y += Integer.parseInt(a[1]);
            }
            else if (a[0].equals("S")) {
                y -= Integer.parseInt(a[1]);
            }
            else if (a[0].equals("E")) {
                x += Integer.parseInt(a[1]);
            }
            else if (a[0].equals("W")) {
                x -= Integer.parseInt(a[1]);
            }
        }
        System.out.println(x + " " + y);
    }
}

File: 240729/방향에 맞춰 이동/move-in-direction.py
dx, dy = [1,0,-1,0],[0,-1,0,1]
x, y = 0, 0
n = int(input())
for i in range(n):
    a, b = input().split()
    if a == "N":
        a = 3
    elif a == "E":
        a = 0
    elif a =="S":
        a = 1
    elif a=="W":
        a = 2
    for j in range(int(b)):
        x += dx[a]
        y += dy[a]
print(x, y)